dogfooding/lib/core/repos/token_service.dart

```diff
@@ -66,13 +74,27 @@ class TokenResponse {
 }
 
 class TokenService {
-  const TokenService();
+  final CustomEnvironmentLoader? customEnvironmentLoader;
+
+  const TokenService({this.customEnvironmentLoader});
 
   Future<TokenResponse> loadToken({
     required String userId,
     required Environment environment,
     Duration? expiresIn,
   }) async {
+    if (environment == Environment.custom) {
+      if (customEnvironmentLoader == null) {
```

Comment:
I think I would drop `Environment.custom`, because now people have to change 2 places to use a custom environment. I would just replace these 2 `if`s with
```suggestion
    if (customEnvironmentLoader != null) {
```
And remove the new custom env.

Reply:
Users don't need to change anything here. The custom environemnt will appear in an env switcher on the login page. So they can choose `custom` there and then their custom loader will be used. Otherwise they can still use our ends if they want to